### PR TITLE
 internal: deploy to the sever via ssh

### LINKED ISSUE #418: Add deploy branch for deploying server
How to run remote ssh commands:
https://github.com/skyrim-multiplayer/skymp-master-api/blob/86c89688dda75870125c94ef313c36c79559a56e/.github/workflows/deploy.yaml
```
docker rm -f ...
docker run
```

If we split docker files into two we can for example split by this line
https://github.com/skyrim-multiplayer/skymp/blob/main/Dockerfile#L62

We should be able to specify any set of branches.
Docker containers are named like branches are.,

### PR DESCRIPTION
**Pull request by nic11**
Closes #418

### CODE REVIEW

File: ci/deploy/push_branch_to_server.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+remote_branch_dir="skymp-server-$DEPLOY_BRANCH"
+
+run_remote test -e "$remote_branch_dir" \
+  || (echo "no branch on remote server" && exit 1)
+
+# FIXME(#164): temporary workaround for Chakra build bug
+cp build/vcpkg_installed/x64-linux/bin/libChakraCore.so build/dist/server/
+cp ci/deploy/workaround_temporary/run.sh build/dist/server/
+
+cp skymp5-server/{package.json,yarn.lock} build/dist/server/
```
Comment (nic11) [inline ci/deploy/push_branch_to_server.sh:54]: Created #613

File: ci/deploy/push_branch_to_server.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+remote_tmp_dir="/tmp/skymp_deploy_`cat /dev/urandom | tr -cd 'a-f0-9' | head -c 32`"
+remote_branch_dir="skymp-server-$DEPLOY_BRANCH"
+
+run_remote test -e "$remote_branch_dir" \
+  || (echo "no branch on remote server" && exit 1)
+
+# FIXME(#164): temporary workaround for Chakra build bug
+cp build/vcpkg_installed/x64-linux/bin/libChakraCore.so build/dist/server/
+cp ci/deploy/workaround_temporary/run.sh build/dist/server/
```
Comment (nic11) [inline ci/deploy/push_branch_to_server.sh:52]: #164

File: ci/deploy/remote/pull_branch.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+branch="${1:?}"
+
+cd "skymp-server-$branch"
+docker stop "skymp-server-$branch" || true
+cp ./server-settings.json server/
+docker run -d --rm --name="skymp-server-$branch" --network=host \
+    -v "$PWD/server:/work" --workdir=/work \
+    -u "`id -u`:`id -g`" \
+    --cpu-period=50000 --cpu-quota=25000 \
+    skymp/skymp-vcpkg-deps ./run.sh
```
Comment (nic11) [inline ci/deploy/remote/pull_branch.sh:15]: Using vcpkg-deps image here for now, will be improved in #584

**Comment by nic11** [comment]
I think everything is up to date now

**Review by Pospelove — APPROVED** [review]

**Pull request merged**